pkg/fqdn/dnsproxy/udp.go

```diff
@@ -176,6 +176,10 @@ func (f *sessionUDPFactory) ReadRequest(conn *net.UDPConn) ([]byte, dns.SessionU
 	return s.m, s, err
 }
 
+func (f *sessionUDPFactory) ReadRequestConn(conn net.PacketConn) ([]byte, net.Addr, error) {
+	return []byte{}, nil, fmt.Errorf("ReadRequestConn is not supported")
```

Comment:
nit: `fmt.Errorf` is for creating formatted errors. When the error message is constant, use something like `errors.New()` instead.
```suggestion
	return []byte{}, nil, errors.New("ReadRequestConn is not supported")
```